### ISSUE TITLE
More than 600 MB Memory Usage

### ISSUE THREAD

**Issue by Andrewsuares** [a85/POSTMan-Chrome-Extension#6015]
Postman is a resource hog on Windows 8.1. Is slow, consumes huge amounts of RAM.

Version
7.0.3 on Windows

**Comment by numaanashraf** [comment]
@Andrewsuares Are you noticing performance issues while performing specific actions?

**Comment by Andrewsuares** [comment]
![capture_001_06032019_105337](https://user-images.githubusercontent.com/35380745/53864213-ea3d0500-4011-11e9-867d-0f9150fdd559.jpg)

I see the performance issue when it is running doing nothing. Yes there is some lag when clicking around collections or clicking on Send. Check out the attached. Returning back to Postman(which is minimized) after doing other things takes time for it to load again.

**Comment by numaanashraf** [comment]
@Andrewsuares Could you share your system details? OS, CPU, Memory & HDD.

**Comment by Andrewsuares** [comment]
![capture_001_07042019_104518](https://user-images.githubusercontent.com/35380745/55679080-84021580-5922-11e9-8a9c-9607d1d779d5.jpg)

Windows 8.1
Intel i3 6th generation CPU
1TB HDD
4GB RAM
Postman consumes too much memory

**Comment by mstaalesen** [comment]
Any update on this? Ever since moving over to Electron Postman is so slow it's making my team seriously consider other options. Postman is no longer the "easy to use" tool that it once was. :(

**Comment by Andrewsuares** [comment]
Postman needs to do a code-rewite of the desktop client. My computer has 12 GB RAM. The start-up time, UI lag and overall performance is ridiculous.

**Comment by mstaalesen** [comment]
Agreed. It can't be unknown for them either, as it is rather easy to reproduce. 

Just have > 10 collections and a few hundred requests in the application. Combine that together with relying on environmentvariables it takes 2-5 seconds just to open a new request. +5 seconds to send the request (that the service uses ~20ms to process) and often 2-5 seconds to render/show the response (if it is more than 1k lines in the response).

Anyone have any suggestions on how to get around this? Clearing environment-variables often? And only having 1 collection open in Postman maybe?

**Comment by preethammavin** [comment]
Hey, this is something that we are actively tracking and trying to improve over the next few releases as this is a key aspect when it comes to delivering the right experience for our users. @mstaalesen the numbers that you have posted are definitely not what we expect, will keep the thread updated as and when we push improvements on performance updates. We recently pushed a release which optimised the Collection runner and we will continue to improve overall Application performance to ensure better experience. 

@Andrewsuares @mstaalesen Just to understand the numbers a bit more, does opening a request take 2-5 seconds every time? Or happens when you sign-in. And would like to know how many collections (ball park number) are present in the team as a whole and how many team workspaces are you a part of?

**Comment by mstaalesen** [comment]
It happends every time. If i clear my environment-variables, it will be quicker for a few requests. And then it gets slow again (since i have a lot of environment variables).

Because of this, I can only have 1 environment in Postman. If i change to several environments, it gets even slower.

I am on Linux btw.

**Comment by kamalaknn** [comment]
Hey @mstaalesen 

Could you give us some more information regarding your environment?

1. How many environment variables do you have?
2. How big is the total size of the environment? (you can export the environment as JSON and get the character length, we just want a rough estimate)

Also, can you tell us the version of Postman you are on? We pushed some performance improvements for response rendering in 7.3

**Comment by elstari-marc** [comment]
> Could you give us some more information regarding your environment?

I'm using the latest version; right now that's 7.9. On Windows 10, 64 bits, with latest update an empty environment, without any requests or variables it already starts with 600 MB memory usage.

Embarrassing.

I know it's Electron's fault, but please fix this. I'm not going to keep using Postman this way. Switch to an other programming tool with which you can produce a more elegant app.

**Comment by mstaalesen** [comment]
Sorry, missed this commend @kamalaknn 

1. right now it is 248 env variables in 1 environment file. 
2. the environment file:
> wc feedback_env.json 
 1248  2000 19323 feedback_env.json

so rougly 20kb in json?

I am on version 7.10.0 on Linux ubuntu 1604. 

Sorry to say, but the performance is still bad and we are looking at replacing Postman with another tool in my team. Good luck in the future!

cc: @a85 @numaanashraf

**Comment by hugoblanc** [comment]
Recently moved from windows to ubuntu and it seems even worse on linux ... Also seriously considering moving to other alternatives ...

**Comment by elstari** [comment]
@preethammavin @kamalaknn 
Can you give us a status update of the progress or planning on this? 

Right now not only the memory usage is outrageous but I have also experienced Postman processes silently running at 75% CPU after I closed the app.

Although real native apps are the way to go (imho), using web techniques for a "native" app can be done without these problems. For example look at Brackets.

If this will not be fixed I think I need to step over to HttpMaster (Windows) and Paw (macOS); all two fully native. Only disadvantage of them is that I can't share projects cross-platform. That's the only tiny reason to stay with Postman for a while ...

**Comment by kukirokuk** [comment]
The same high RAM usage on Ubuntu 18.04.

**Comment by Reez0** [comment]
I'm experiencing something similar. Not with RAM usage, but with CPU Usage.

![Untitled3](https://user-images.githubusercontent.com/28865166/75327642-6e247880-5885-11ea-91bd-1880288fdb0f.png)

It fluctuates between 50% and 60%. 

Windows 10 Pro
Intel Core i5 7200-U
8GB Ram
Postman Version 7.19.1

I have a single collection containing 14 requests.

**Comment by mccannt** [comment]
Is this still an issue for anyone in the thread? We have made many performance improvements to the application recently. 
Can you please try the most recent version of the Postman app and let us know if this is still an issue?

**Comment by Andrewsuares** [comment]
Version 7.30.1 is still slow. 300 - 350 MB RAM consumed doing nothing. About 100 API requests inside collections. 15 seconds of initial start time spent on looking at the spinner.

Windows 8.1, 12 GB RAM.

**Comment by mccannt** [comment]
Thanks for letting us know @Andrewsuares and we are sorry for your experience.
We are always looking at improving the performance of the application so please keep updated to the most recent version.

We will keep you updated as things progress.

**Comment by dave-gil** [comment]
This is something I'm regularly experiencing using latest v8.9.1 on Windows 10 x64 with a decent spec machine i7 2.7GHz wtih 16GB ram:

For example while running a very simple collection with 200-250 requests each returning only approx 100Kb per request. Postman memory usage balloons to 3gb and Postman crashes. Is there anyway to prevent or limit this accumulation of memory usage (in Windows). 

I'm using no environment variables and only a few collection variables (non of which are memory intensive)

Why does Postman memory usage balloon so high? It seems to consecutively increase with each request during the running of the collection?

**Comment by Neutrino-Sunset** [comment]
Still have the same issues with v10.14

I run a medium sized collection of fast running tests with smallish payloads, all of which pass.

During the collection run itself resource usage is not unreasonable.

About 30 seconds after the collection run completes Postman CPU usage starts maxing out on one core and it starts eating up to 1.8 Gig of RAM. Before I switched off Hardware Acceleration it was using over 2.5 Gig. It then stays in this state for a couple of minutes during which time the system is not really usable just because of the sheer amount of noise it's making.

Even during normal operation Postman runs really slow for all team members, duplicating requests, importing and exporting collections and such like taking 30 seconds and upward.

We are also starting to evaluate alternative tools as the hit to productivity is getting to be more than is acceptable, but it would be good if this could be fixed.

This is on Windows 10.

![Postman](https://github.com/postmanlabs/postman-app-support/assets/18445931/20bff643-1645-4d15-a6e8-310d9f49b0ba)

**Comment by MC-sanderv** [comment]
Closing the issue due to being stale. 
Peformance issues for Postman is a common issue for some users and newer similar reports have been also opened since this issue. 
This issue also happened on old Postman versions, we have had many Electron upgrades and Postman improvements since then. 


Should this continue to be an issue, please open a new ticket with relevant information.
